File: php/features.php
<?php
declare(strict_types=1);

// Magic8Ball SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class Magic8BallFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new Magic8BallBaseFeature();
            case "test":
                return new Magic8BallTestFeature();
            default:
                return new Magic8BallBaseFeature();
        }
    }
}

File: py/features.py
# Magic8Ball SDK feature factory

from feature.base_feature import Magic8BallBaseFeature
from feature.test_feature import Magic8BallTestFeature


def _make_feature(name):
    features = {
        "base": lambda: Magic8BallBaseFeature(),
        "test": lambda: Magic8BallTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
